.github/actions/setup-go-tip/action.yml

```diff
@@ -8,16 +8,38 @@ runs:
       shell: bash
       run: |
         set -euo pipefail
-        tip=$(git ls-remote https://github.com/golang/go.git HEAD | awk '{print $1;}')
-        echo "Go Tip version: ${tip}"
-        curl -fsSL https://storage.googleapis.com/go-build-snap/go/linux-amd64/${tip}.tar.gz -o gotip.tar.gz
-        echo "Downloaded bundle:"
-        ls -lah gotip.tar.gz
-        export GOROOT="$HOME/sdk/gotip"
-        mkdir -p $GOROOT
-        tar -C $GOROOT -xzf gotip.tar.gz
-        export PATH="$GOROOT/bin/:$PATH"
-        echo "GOROOT=$GOROOT" >> $GITHUB_ENV
-        echo "PATH=$PATH" >> $GITHUB_ENV
-        echo "Active Go version:"
-        go version
+        retries=10
+        wait_time=30
+        success=false
+        for ((i=1; i<=retries; i++)); do
+          tip=$(git ls-remote https://github.com/golang/go.git HEAD | awk '{print $1;}')
+          echo "Go Tip version: ${tip}"
+          if curl_output=$(curl -fsSL -w "%{http_code}" -o gotip.tar.gz https://storage.googleapis.com/go-build-snap/go/linux-amd64/${tip}.tar.gz); then
+            http_code=$(tail -n1 <<< "$curl_output")
+            
+            if [[ "$http_code" == "404" ]]; then
+              echo "HTTP 404: File not found"
+            else
+              echo "Downloaded bundle:"
```

Comment:
this could be some other failing HTTP code, `not 404` does not mean success

Reply:
yes, another thing i noticed now if `curl` returns error code then the nested `if` will never execute
```bash
if curl_output=$(curl -fsSL -w "%{http_code}" -o gotip.tar.gz https://storage.googleapis.com/go-build-snap/go/linux-amd64/${tip}.tar.gz); then
  http_code=$(tail -n1 <<< "$curl_output")
  if [[ "$http_code" == "404" ]]; then
    echo "HTTP 404: File not found. Retrying in $wait_time seconds..."
    sleep $wait_time
```
I don't think we need to check for error code `404` specifically